### PR TITLE
build: Migrate ts-loader to esbuild-loader

### PR DESCRIPTION
**Pull request by JadenSimon**
## Description
Replaced ts-loader and Terser with esbuild-loader to see if we get any sort of increase in build-speed. This PR requires the extension to be transpiled to es2018 instead of es2015 (ES6) because esbuild does not support async generators and async iterators. Changing the build target shouldn't affect anything since VSCode appears to support it. 

Note that webpack is still being used. Esbuild-loader merely acts as a wrapper for esbuild so it can be used with webpack. Webpack could be removed entirely but it would require a much larger change in the build process.

Three unique warnings currently appear (not sure if they appeared before, will have to look into this...)

`WARNING in ./src/lambda/config/templates.ts 114:9-34
"export 'ParseErrorCode' (imported as 'jsonParser') was not found in 'jsonc-parser'
 @ ./src/shared/sam/debugger/commands/addSamDebugConfiguration.ts
 @ ./src/shared/sam/activation.ts
 @ ./src/extension.ts
`

`WARNING in ./node_modules/typescript/lib/typescript.js 2710:20-42
Critical dependency: the request of a dependency is an expression
 @ ./src/shared/typescriptLambdaHandlerSearch.ts
 @ ./src/shared/codelens/typescriptCodeLensProvider.ts
 @ ./src/shared/sam/activation.ts
 @ ./src/extension.ts
`

`
WARNING in ./node_modules/adm-zip/util/fileSystem.js
Module not found: Error: Can't resolve 'original-fs' in '/Users/sijaden/VSCode/aws-toolkit-vscode/node_modules/adm-zip/util'
 @ ./node_modules/adm-zip/util/fileSystem.js
 @ ./node_modules/adm-zip/util/index.js
 @ ./node_modules/adm-zip/adm-zip.js
 @ ./src/eventSchemas/commands/downloadSchemaItemCode.ts
 @ ./src/eventSchemas/activation.ts
 @ ./src/extension.ts
`

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by justinmk3** [comment]
Looks like a 30% speedup, comparing `VSCode Package Test` times: 3min on master vs 2min in this PR.

**Comment by JadenSimon** [comment]
Yeah pretty good gains for a simple change. I'd imagine removing webpack entirely would be more substantial.

File: webpack.config.js [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 const path = require('path')
 const webpack = require('webpack')
+const { ESBuildMinifyPlugin } = require('esbuild-loader')
 const fs = require('fs')
 const TerserPlugin = require('terser-webpack-plugin')
```
Comment (justinmk3) [inline webpack.config.js:14]: can this= be removed ?

Comment (JadenSimon) [reply webpack.config.js:14]: Yeah should be able to remove it, I don't see it being used elsewhere.

**Comment by hunterwerlla** [comment]
`This PR requires the extension to be transpiled to es2018 instead of es2015 (ES6) because esbuild does not support async generators and async iterators. Changing the build target shouldn't affect anything since VSCode appears to support it.`
Same issue with azdo, is supported on node10+ so just echoing this should have no issues

File: webpack.config.js [hunk 2]
```diff
@@ -58,11 +58,10 @@ const config = {
                         },
                     },
                     {
-                        loader: 'ts-loader',
+                        loader: 'esbuild-loader',
```
Comment (justinmk3) [inline webpack.config.js:61]: looks like we can also remove `ts-loader` dependency!

**Comment by justinmk3** [comment]
> `WARNING in ./src/lambda/config/templates.ts 114:9-34 "export 'ParseErrorCode' (imported as 'jsonParser') was not found in 'jsonc-parser' @ ./src/shared/sam/debugger/commands/addSamDebugConfiguration.ts @ ./src/shared/sam/activation.ts @ ./src/extension.ts `

I suspect this warning is from dead code elimination. It looks like we don't import `templates.ts` anywhere:

    git grep! -E 'import.*\btemplates[^/]\b'

**Pull request merged**